File: plannerApp/src/models/invite.ts
import { Family } from "./family";
import { User } from "./user";

export interface Invite {
    id?: number;
    fromUserUid: string;
    fromUser?: User;
    toUserUid: string;
    toUser?: User;
    status?: string;
    toFamilyId: number;
    family?: Family;
}

export const inviteStatus = {
    Pending: "Pending",
    Received: "Received",
    Accepted: "Accepted",
    Declined: "Declined",
}

File: backend/models/InviteModel/Invite.cs

using FamilyPlanner.Models.FamilyModel;
using FamilyPlanner.Models.UserModel;

namespace FamilyPlanner.Models.InviteModel {
    public class Invite {
        public int Id { get; set; }
        public string FromUserUid { get; set; }
        public User? FromUser { get; set; }
        public string ToUserUid { get; set; }
        public User? ToUser { get; set; }
        public string? Status { get; set; }
        public int ToFamilyId { get; set; }
        public Family? Family { get; set; }
    }
}